### PR TITLE
Feature/f 4405

### PR DESCRIPTION
**Pull request by saralabafi**

### CODE REVIEW

File: services/services.ts [hunk 1]
```diff
@@ -17,6 +18,25 @@ const GetData = async ({ queryKey }: QueryFunctionContext) => {
   return res.data
 }
 
+const UpdateData = async (data: any) => {
+  const pageContent = { Widgets: data }
+  const response = await fetch(
+    'http://gateway.dourtal.co/api/cms/v1.0/{site}/pages/ca320982-04cf-47dc-233c-08dbb45cb49a',
+    {
```
Comment (amirkh330) [inline services/services.ts:25]: It must be dynamic and we get route + id

File: app/[locale]/page/Layout.biz.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export const useLayout = () => {
   const { data } = useQuery(
     [
       {
-        url: 'cms/v1.0/{site}/pages/5E155016-EDC0-4364-D0D9-08DBB1CFC585',
+        url: 'cms/v1.0/{site}/pages/ca320982-04cf-47dc-233c-08dbb45cb49a',
```
Comment (amirkh330) [inline app/[locale]/page/Layout.biz.ts:17]: I think we have to use a URL for reading the ID

File: app/[locale]/page/Layout.biz.ts [hunk 3]
```diff
@@ -46,3 +52,18 @@ const processData = (data: any, parentId = 0) => {
     return newItem
   })
 }
+const modifyData = (designList: any) => {
```
Comment (amirkh330) [inline app/[locale]/page/Layout.biz.ts:71]: we have Control type, could we use it?

File: services/services.ts [hunk 4]
```diff
@@ -17,6 +18,25 @@ const GetData = async ({ queryKey }: QueryFunctionContext) => {
   return res.data
 }
 
+const UpdateData = async (data: any) => {
+  const pageContent = { Widgets: data }
+  const response = await fetch(
+    'http://gateway.dourtal.co/api/cms/v1.0/{site}/pages/ca320982-04cf-47dc-233c-08dbb45cb49a',
```
Comment (amirkh330) [inline services/services.ts]: It must be dynamic and we get route + id

File: app/[locale]/page/Layout.biz.ts (hunk 2 above)
Comment (amirkh330) [inline app/[locale]/page/Layout.biz.ts]: I think we have to use a URL for reading the ID

File: services/services.ts [hunk 5]
```diff
@@ -17,6 +18,25 @@ const GetData = async ({ queryKey }: QueryFunctionContext) => {
   return res.data
 }
 
+const UpdateData = async (data: any) => {
+  const pageContent = { Widgets: data }
+  const response = await fetch(
```
Comment (amirkh330) [inline services/services.ts]: fetch?

File: app/[locale]/page/Layout.biz.ts (hunk 3 above)
Comment (saralabafi) [reply app/[locale]/page/Layout.biz.ts:71]: No, When we use Control we have errors in body of map.

File: app/[locale]/page/Layout.biz.ts [hunk 6]
```diff
@@ -1,20 +1,26 @@
...
 
 export const useLayout = () => {
   const { designList, activeControl } = useSelector(
     (state: RootState) => state.pageDesign
   )
   const dispatch = useDispatch()
+  const router = useRouter()
+  const id = 'ca320982-04cf-47dc-233c-08dbb45cb49a'
+  const searchParams = useSearchParams()
+  console.log(searchParams.get('id'))
```
Comment (amirkh330) [inline app/[locale]/page/Layout.biz.ts]: remove console.log

File: services/services.ts [hunk 7]
```diff
@@ -18,20 +18,17 @@ const GetData = async ({ queryKey }: QueryFunctionContext) => {
...
-    'http://gateway.dourtal.co/api/cms/v1.0/{site}/pages/ca320982-04cf-47dc-233c-08dbb45cb49a',
-    {
-      method: 'PUT',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(pageContent),
-    }
-  )
+  const response = await base_api.put(`cms/v1.0/{site}/pages/${id}`, {
```
Comment (amirkh330) [inline services/services.ts]: we must write updateData func that every person can use by send dynamic route not unique for one point

**Pull request merged**